### PR TITLE
[Finally Finished] Toxins Techweb Pointcap

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl: Cobby (using cyclown's formula)
balance: Techweb point gen via toxins are now very-hard-soft capped at 50K point rewards. After 50K points have been awarded, you can still get points for bombs that would reward higher than 50K, but they are worth only a flat 1K (hint: it's not beneficial). Give those other bombs to mining!
balance: Techweb toxins equation has been edited (once again)
/:cl:

Reason: 10 minute prep is not worth nullifying the entire balancing act of techwebs, which are meant to more/less dripfeed items into an hour-hour thirty round.

Constructive feedback is appreciated regarding the math, particularly from @kevinz000 .

### CODE REVIEW

**Comment by optimumtact** [comment]
label ur fukken axes

**Comment by ExcessiveUseOfCobblestone** [comment]
fine...

**Comment by RogueSteampunker** [comment]
50k for basically bootlegging a proper tritium setup from a unupdated toxin lab, which won't even cover the people who rush nanite tech every round.

look, I get the fact that getting all the tech roundstart is silly. But considering the fact that a round that lasts an hour and a half only happens on lowpop Sybil or extended, it seems silly to nerf it as if we have 3 hour rounds. And once again, we have practively nothing that science can do to help gen points, a la engineers doing nothing.

**Comment by RogueSteampunker** [comment]
50k for basically bootlegging a proper tritium setup from a unupdated toxin lab, which won't even cover the people who rush nanite tech every round.

look, I get the fact that getting all the tech roundstart is silly. But considering the fact that a round that lasts an hour and a half only happens on lowpop Sybil or extended, it seems silly to nerf it as if we have 3 hour rounds. And once again, we have practively nothing that science can do to help gen points, a la engineers doing nothing.

**Comment by ExcessiveUseOfCobblestone** [comment]
According to the [Duke's RND Guide made by a girl 
](https://docs.google.com/spreadsheets/d/1OD4wQAI5X8inAPJIwVxY8j8PvnKNHWLgXwtdORWnfeQ/edit#gid=1934749195)
NANITE | 36,000 | 0:11:07
-- | -- | --

**Comment by ninjanomnom** [comment]
The visual is actually this. I modified it a bit after screenshotting that earlier one to accurately hit the point cap in the same place.
image

File: code/game/machinery/doppler_array.dm [hunk 1]
```diff
@@ -114,12 +114,14 @@ GLOBAL_LIST_EMPTY(doppler_arrays)
...
-		else
-			point_gain = round(((log(x)-1)**1.6)*TECHWEB_BOMB_POINTCAP)
+	
+	/*****The Point Calculator*****/
+	
+	if(orig_light < 10)
+		say("Explosion not large enough for research calculations.")
+		return
+	else
+		point_gain = round(((log(x)-1)**1.6)*TECHWEB_BOMB_POINTCAP)
```
Comment (ninjanomnom) [inline code/game/machinery/doppler_array.dm]: x was the graphing var not the real one

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by optimumtact** [comment]
bombs have long lacked purpose and fucking up the rest of the game just to give "toxins" something to do isn't a good way to do it cyclowns

File: code/game/machinery/doppler_array.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/machinery/doppler_array.dm]: lol

**Comment by ExcessiveUseOfCobblestone** [comment]
The problem with toxins/atmos is that it becomes a litmus test after you know how to do something. It's also quite calculator easy if you cared enough (since you code a lot of atmos stuff i'm sure you've done this already, hence why you're able to tout you can make the biggest bombs).

This PR still benefits people who can make larger bombs (which as of now involves watching a youtube video to complete) while also being a neat side-benefit for people who want to explore the mechanic. More importantly, it no longer forgoes our progression system.

The issue with toxins needing to do something is not my problem, as it has existed long before techwebs. If it's only found use now, and that use is to forgo the feature, it needs serious review.

**Comment by cyclowns** [comment]
@optimumtact i completely agree with you, which is why i proposed an alternative solution of simply reducing the values in the original function instead of reducing toxins to what is essentially uselessness. just because bombs have 'long lacked purpose' doesn't mean they shouldn't have purpose.

@ExcessiveUseOfCobblestone "The problem with toxins/atmos is that it becomes a litmus test after you know how to do something. "

and other departments don't? the only reason toxins and atmos' version of a litmus test has manifested itself as a problem is because toxins bomb points are by far the best choice for point generation. no one is complaining that SM minmaxing for rad collector points is a 'litmus test' because it's fucking useless.

**Comment by 81Denton** [comment]
I think that the better timing for this would be after @kevinz000 introduces new methods for point generation that (IIRC?) he's working on.

Ballmer peak is miniscule and inconsistent; tesla coil research is only feasible on Pubbystation and rad collector research is so insignificant that it's only seen on downstreams with 2+ hour rounds.

**Comment by Tlaltecuhtli** [comment]
can we start make techs use different type of points so like example toxin bombs help weapon related branches instead of cobble doing  50 prs until we revert back to old rnd

**Comment by ExcessiveUseOfCobblestone** [comment]
> and other departments don't? the only reason toxins and atmos' version of a litmus test has manifested itself as a problem is because toxins bomb points are by far the best choice for point generation. no one is complaining that SM minmaxing for rad collector points is a 'litmus test' because it's fucking useless.

The point being? We've had features that haven't became an issue for lengths after their addition until 1/few people start using it consistently to debatable abusive natures. The problem with this feature in particular is that this calculation seems to encourage that. In other departments, they have been balanced with that in mind. Knowing how to do genetics/xenobio doesn't ensure I get massive golem spam or powers within a set time. Knowing how to do botany doesn't ensure I get good Max Yield Max Potency plants within a set time. Atmos (and toxins) is very controlled if you know what you're doing.

Maybe if there was something other to do with bombs that would make you have to choose between an epic amount of points and the other thing then this would be alright. As it stands its ONLY use (besides leveling the station) is to make points.

> people will just make 50k point bombs in 5 minutes and then fuck off for the rest of the shift

In addition to Grazyn's comment, I think that speaks more volumes about toxins than it does pointgen.

> This PR still benefits people who can make larger bombs

What I mean is that you still get epic points for making larger bombs than those that make smaller bombs. No, it is no longer going to benefit you as much for making a bomb larger than 50K worth of points, that's the entire point of this PR.

> why are you presenting people putting legitimate effort into teaching others how to get into an obscure and often misunderstood part of the game as an issue?

Because the feature is implicitly balanced around "not everyone knows how-to so I can make it worth more!".

**Comment by ExcessiveUseOfCobblestone** [comment]
Travis is dumb please ignore her majesty.

**Comment by kevinz000** [comment]
ANYWAYS:
I've recently experimented with fusion and thanks to some help from Joe Berry I'm probably now able to make a different type of bomb that uses fusion products to heat it (or maybe even outright fusion reactions in the TTV, but I haven't been able to have time to make one for a PoC yet).

radius = light explosion
My idea's modify the equation so it goes up to 10-15k for a regular plasma-oxygen bomb that gets like 22/44/88, when it goes higher than 100 the equation changes to give up to 70k for tritium bombs (50k for a tier 2 one that takes 10 minutes (radius 100-1200), maybe 60-70k for one that takes longer and uses tier 3 coolers (radius 1200-2200)).
Fusion bombs, however, generally take atmospherics cooperation to do and takes far longer in comparison, to do the fusion, get the high temperature hyper nobelium/plasma/whatever you're using, get the enormous amounts of tritium/oxygen to put in the cold tank, etc etc. Those can range from 40000-80000 theoretical radius. Give those the old 100k-200k scaling, since it requires cooperation (or atleast someone with access to atmos), and a lot of time to do so.

Why: Plasma-oxygen bombs are the "lowest effort" ones, requiring only heater/coolers, so they shouldn't give much. Tritium requires a bit more knowledge and effort, although not much even for tier 3 (which is the highest you can go). Fusion-required ones (the proven one being using fusion temperatures to boost a tritium bomb so yhou can stuff more tritium/oxygen in it with colder cold tank tempratures) require far more effort (and fusion is quite lethal too) so should be more rewarding.

**Comment by epochayur** [comment]
where the fuck are you playing rounds that go for 90 minutes

**Comment by Whoneedspacee** [comment]
Adding onto kevinz whole post, I think a 1/3 multiplier for the tachyon would be fine.

Reasoning for this, the best possible bomb before hypernoblium which takes a lot of time and fusion to acquire is only about 250k to 300k. A 1/3 multiplier would make it so bigger bombs are still rewarded, but you can only make about 100k points max.

If you’re still mad about 100k then just make it 1/4 for 75k and so on. There’s no need to dumb down toxins to a 50k flat because then I’ll just release a video of my tank bomb making 50k in 2 minutes.

Yes there should be a nerf, yes this is an awful way to do it in this state of toxins.

Also a fix for rad collector generation would be nice, a more constant gain would be much better than a big jump like toxins bombs are, and it’s a constant thing you have to upkeep.

Rad collectors need to update the points per minute they give to the research consoles, and need to be vastly uptuned. They generate a few points every 2 seconds at the best, so the best you can do is like 100 points a minute with every single collector running it.

**Comment by Tlaltecuhtli** [comment]
yeah but by the time you do big bombs you have done like 80% of techwebs by passive generation, why not add like some bs 200K points technodes that unlocks bycicles n shit

**Comment by ExcessiveUseOfCobblestone** [comment]
I think my issue regarding the larger point amounts is that it becomes a forced portion of balance that becomes, quite frankly, a real pain in the arse particularly for late game content.

"Techwebs is balanced around the 1.5-2 hour mark UNLESS someone does toxins, in which case techwebs is _balanced_ around thirty-fourty minute mark UNLESS both atmos and toxins know what they're doing, in which case techwebs is _balanced_ around the twenty-thirty minute mark".

These items are powerful because the implication is it won't be every round, that's literally the point of techwebs. What happens when the same people (from Sybil I'm thinking Harold Robinson) or Kekhi Anderson do it every round out of what they feel to be a necessity?

One side issue is that having a one-stop shop means there's no incentive for people to 
1) Do other items (the obvious response to that is well there really isn't any other way)
2) Code other items to do (doubly so when our active coders happen to be people who play science quite frequently)

**I think setting a hard ceiling for each job encourages the idea that each department should play a role in the progression as opposed to (surprise surprise) just science being the powerhouse for RND. If capping it frustrates players, I hope that consequently encourages them to idea-guy or even code new avenues for other roles to assist.**

**Comment by ExcessiveUseOfCobblestone** [comment]
> There’s no need to dumb down toxins to a 50k flat because then I’ll just release a video of my tank bomb making 50k in 2 minutes.

Was getting caught part of your plan?

**Comment by Whoneedspacee** [comment]
The point of my post was just remediating what is already there so most bombs would be useful, if you wanted to add something to make hypernoblium bombs something to chase after that’s fine.

I created a literal speedrun route and the earliest I could do it on bagil without anyone bothering me was 30 minutes since it requires fusion of at least mid tier level, about 3600 moles of cooled tritium and oxygen down to 2.8 kelvin, then using the fusion to heat up a mix to make hypernoblium, then tanking both of those.

**Comment by RogueSteampunker** [comment]
"I think setting a hard ceiling for each job encourages the idea that each department should play a role in the progression as opposed to (surprise surprise) just science being the powerhouse for RND. If capping it frustrates players, I hope that consequently encourages them to idea-guy or even code new avenues for other roles to assist"

Eeeexcept that, other than Kev adding the infrastructure for alternative point values to be added, nobody has tried anything of worthwhile value, even when toxins floundered during the great atmos change of a couple weeks back, when regular bombs did jackshit and tritium came by to pave the way. Seriously, adding code that'll make current tasks even MORE of a chore without actual changes lined up is how we make something ripe for removal rather than improvement, a la experimentor.

**Comment by Whoneedspacee** [comment]
Toxins is more fit to be an atmos / engineering department job anyways. The actual science department for toxins is just inferior in every way to something like an atmos incinerator setup with hundreds of thousands of moles at your disposal.

And yes if you keep making techwebs a pain in the ass someone is bound to change them, it's a bad system when people don't find it fun, and what you'll end up doing is making nobody play science and just wait 40 minutes until you can actually research shit. You need to add something else for point generation at the same time as nerfing point generation in other ways, otherwise it provides an imbalance that might not be fixed for months and makes everything shitty.

Also I really don't like how the current system inspires greedy researching, what people will do is make a bomb in 2 minutes, then blow it up and research everything they want without anyone else being able to do anything. All this is going to inspire is a broken system for powergamers to make mechs unless you actually make some other methods for point generation.

**Comment by cyclowns** [comment]
>Techwebs is balanced around the 1.5-2 hour mark UNLESS someone does toxins, in which case techwebs is balanced around thirty-fourty minute mark UNLESS both atmos and toxins know what they're doing, in which case techwebs is balanced around the twenty-thirty minute mark

the most confusing part of this sentence is the assertion that techwebs is balanced around 1.5-2 hour rounds when bagil/sybil rarely even go above 45 minutes

**Comment by Grazyn** [comment]
"Softcap" should be 0, otherwise scientists will continue to make the streamlined bomb roundstart because starting with 50k points is still better than nothing, as shown by whoneeds.

Starting with free points is still bad for techwebs, be it 200k or 50k because it breaks balance.

**Comment by kevinz000** [comment]
It was originally intended to be balanced to passive gen Max out after 90 minutes.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Grazyn Yes but I get tired of posting and feel obligated to respond because it's my PR so I give in.

@cyclowns https://atlantaned.space/statbus/round.php Bagil runs ~15/~30/~60 minute rounds. Sybil regularly pushes over 2 hours, Terry typically over 4 (jesus I can barely play 2).

The feature, as kevinz said, was marketed to max at around the hour thirty mark hence the comment you quoted.

**Comment by Whoneedspacee** [comment]
>please reframe your thinking

No?

I think it's honestly quite annoying that you seem to think that the players have to abide by your thinking of what is FUN. Also, you seem to imply that toxins is nothing more than ratios and recipes, little skill is required, hmm that reminds me of many jobs like...

Chemistry
Xenobiology
Robotics
Engineering
Medbay

Hmm it's almost like when you learn something it becomes much easier to do, how interesting.  It's quite quite annoying that you insult toxins for being too linear when there's jobs like chemistry which are way more linear and give much stronger stuff than what atmos or toxins give.

Nobody is saying toxins should stay like it is, but you clearly have no idea what you're talking about when you say toxins should be capped, and I doubt your skill in the area. I have posted multiple times that a 50k bomb is easily achievable at round time 8 minutes, it will be just as spammed as it is currently, a 1/4 point multiplier or more would be much better.

**Comment by Grazyn** [comment]
@Whoneedspacee I don't think you understand what Oranges means.
It is true that all jobs have a boring checklist of things to do roundstart before they can start having fun, but bombmaking for techwebs isn't the same thing because techwebs isn't a job. When you make a 50k/100k/200k bomb roundstart you're not doing it for the sake of toxins, you're doing it for tech points. Techwebs were designed to trickle new stuff into the round and passive point generation is basically a timer. If people make a 50k bomb after 2 minutes, which means 20 free nodes, it skews the balance and throws the entire concept in the trash. Multiplier doesn't change anything, it still means that people will do bombs roundstart to start with free points, which clearly wasn't intended when coders (probably unfamiliar with bomb making and all the related powergaming) added bombs as a way to generate points in a single burst.

**Comment by optimumtact** [comment]
Doubt my skill all you want, it won't change reality, which is that I have a very clear design goal for techwebs and this PR assists in that vision.

**Comment by ninjanomnom** [comment]
>I think it's honestly quite annoying that you seem to think that the players have to abide by your thinking of what is FUN

It is in fact exactly headcoder/maintainer's job to decide what's best for the game even if the players don't like it. Feedback and discussion is welcome but keep it civil.

**Comment by optimumtact** [comment]
If you're  serious toxins player you should be happy to see it migrating to atmospherics, that would be a good thing for you.

**Comment by ExcessiveUseOfCobblestone** [comment]
Please don't say archaeology you know that's my trigger word anon

**Comment by Shdorsh** [comment]
Have you tried it with this:

point_gain = TECHWEB_BOMB_POINTCAP * (1 - 10**(-point_gain / TECHWEB_BOMB_POINTCAP))

It should give a nice inverted hyperbole that caps at 50000 and you won't have the problem of those ugly looking low values that suddenly spike up.

**Comment by Whoneedspacee** [comment]
How are you doing log(point_gain), that’s always going to be 0 which is undefined. Did you forget to set it or something?

**Comment by ExcessiveUseOfCobblestone** [comment]
I was trying to sabotage but you've caught me!

Used the wrong var, my bad :^)

**Comment by Whoneedspacee** [comment]
The formula is definitely a step in the right direction, however through previous bomb examples I think the formula needs to be even less points gained for light radius.

Let’s say we have a 43k point bomb before this pr, easily achieveable by round time 5 minutes or less. To calculate light radius we do (43000/3000)^2 which gives about 202. Looking at the graph of your formula that already reaches double maximum point generation and then some.

It’s a step in the right direction but it does need changes still. I personally like (log(x) / log(10000))^1.6 because that means you need a 10000 light radius or a 300k bomb before this pr to max out, which can take much longer.

**Comment by ninjanomnom** [comment]
Why are you doing log() on a constant in that. If you want to scale the formula for higher light levels than 100 cleanly and allow further change of that maximum size without needing to come up with a new formula:
`((log(orig_light)-1)**1.6)*TECHWEB_BOMB_POINTCAP`
can be converted to
`((log(max(TECHWEB_BOMB_SIZECAP*0.1, orig_light)/(TECHWEB_BOMB_SIZECAP*0.01))-1)**1.6)*TECHWEB_BOMB_POINTCAP`

**Comment by Shdorsh** [comment]
Woah, you folks really don't like it simple...

**Comment by ninjanomnom** [comment]
It's just a log curve friend

**Comment by ExcessiveUseOfCobblestone** [comment]
i'm scared at maths

**Comment by Shdorsh** [comment]
If you want something that nicely curves and stops exactly at the cap, just try putting my formula in your spreadsheet and seeing what it does.

**Comment by ninjanomnom** [comment]
The one currently does that since anything higher is already ignored

**Comment by ExcessiveUseOfCobblestone** [comment]
So I put in y = 50000*(1-10^{(-x*5/50000)}) and it told me it would be 20K light radius before I could get 49.5K points, is this what you wanted lol.

**Comment by Shdorsh** [comment]
You can change the value 5 to be something more appropriate

**Comment by octareenroon91** [comment]
A function with a true, guaranteeable cap:
https://en.wikipedia.org/wiki/Logistic_function

**Comment by granpawalton** [comment]
I have a follow up PR for this ready that should address some of the issues i had with this and hopefully some of the issues the other toxins players had with this while keeping the hard cap. I will PR it as soon as this is merged.

**Comment by ExcessiveUseOfCobblestone** [comment]
no just make a counter PR that `closes #40107` so I don't have to deal with the number dancing

**Comment by granpawalton** [comment]
I closed the PR i made because it was unpopular, didnt want to ask for what i wanted without atleast coding it. 

I am still worried about having left over bombs though.

**Comment by Whoneedspacee** [comment]
This PR shouldn't merge without some actual at least theoretical bomb testing to balance it, same with the other PR that was closed. Both of these have mathematical issues because this formula is still cheeseable, all this pr needs is a formula change for the point gain and it will be fine.

**Comment by ExcessiveUseOfCobblestone** [comment]
The point formula being cheesable is because bombs themselves are cheesable, there's no way to make the system not cheesable unless I start overhauling bomb code which is

1) cancer to deal with in comments
2) prone to hecking up something else unexpectedly.

As for leftover bombs, you give them to mining or whatever you used them for prior to techweb's implementation.

**Comment by granpawalton** [comment]
The problem with giving them to mining is that, like i said in my pr, the time to set up and trade it over to the other player could produce the same results by just mining with a plasma cutter for that same amount of time without using up such a powerful tool, and thats assuming noone dies before they can meet up or finds other urgent things to attend to.

There was also the issue of people “accidentally” leaving bombs out for their meta buddies who happen to be traitors which i have seen happen several times, the old system/my pr atleast put a way for the bombs to go anywhere besides that.

Still though i would rather this than the previous system dont get me wrong.

**Comment by Whoneedspacee** [comment]
Bombs themselves are not cheesable and you've been given solutions to your formula. Your formula doesn't work because you didn't base it in reality when you created it / test it at all.

**Comment by cyclowns** [comment]
>"The point formula being cheesable is because bombs themselves are cheesable, there's no way to make the system not cheesable unless I start overhauling bomb code"

sounds like excuses to me for not wanting to design a half-decent formula for bombs

to prove my point i've designed a pretty good bomb point formula based in reality
notes on the methodology used are in the desmos calculator itself

https://www.desmos.com/calculator/3b6uzupffb
image

i suggest that this PR implements this formula instead

**Comment by ExcessiveUseOfCobblestone** [comment]
@granpawalton this PR isn't meant to cure the underlying problems with the "what do I do after I actually make the bombs" job.

My basings of the calculation where 100 is the golden number are for people who are exploring the mechanic casually after the 20 LR which is our maxcap, not people codediving and formulating the perfect storm. I want the difference between 20 to say 23 be something tangible rather than lolnope it because I have guys who've dug into or even coded the process (OOC things) able to make bombs multitudes higher. There's no way for me to balance around both people who are casually exploring the mechanic and diehards who can do 10x+ better and make it meaningful for both, while also considering that the benefits of getting points affect others playing the game. If you think you can, go ahead. **Not considering people can make bombs theoretically in the thousands is not a lazy option, that's a conscious decision to cater to those that are exploring the mechanic casually.** Again, my formula caps it at 100 so that bombs higher than the current capped radius still give tangible amount of points. Formulas that veer off to 50K do not do that until extremely high radii, which will only ever benefit people who codedive (or watched a video from someone who has), something I don't want to encourage.

**Comment by cyclowns** [comment]
>which will only ever benefit people who codedive (or watched a video from someone who has), something I don't want to encourage.

care to explain why codediving in an _open source game_ is evil and should not be encouraged? i'm not saying it should be ENCOURAGED, but reading the code is kind of just like reading a more in-depth and harder to understand wiki article.

are you instead suggesting that everyone should just do everything by hand, in-game, with absolutely zero outside help? because that's how you get people to give up immediately.

>My basings of the calculation where 100 is the golden number are for people who are exploring the mechanic casually after the 20 LR which is our maxcap

lol what?? have you ever made a bomb in the last 6 months? this is even more proof that you didn't base your formula in reality. you can't get to 100 LR without tritium (or a ton of arcane knowledge that no one 'exploring the mechanic casually' would ever find out before giving up) after the plasma fire '''nerf'''. if my plasma buff had been merged instead of dismissed because the nerf would 'shake up the bomb meta' (completely disregarding the fact that tritium had always been the meta since it was released), then your assertion that 100 is a golden number for people exploring the mechanic would be true.

the next step after maxcaps as it stands right now is creating tritium bombs for research. figuring out how to make trit casually is _NOT THAT DIFFICULT_, even without codediving. the next step after that is figuring out how to maximize trit output and combine it with plasma to create plasma-oxy/trit bombs, which are still _ABSOLUTELY POSSIBLE FOR SOMEONE TO FIGURE OUT HOW TO MAKE CASUALLY_. it might take them a lot of testing and experimentation to do so (or maybe.. you know.. asking another player how they make large bombs, which according to you is some taboo evil), but as long as you have some time to waste and more than three brain cells its possible.

you might notice that coincidentally, the formula i presented is based on the actual state of toxins that i just outlined!

**Comment by granpawalton** [comment]
Im not asking you to fix the underlying issue, im asking that you not make it worse without attempting to patch any damage youve done.

Also as we can see on our current player base you dont have to be smart or a code diver to make a normal bomb or trit bomb which is exactly what cyclowns formula is designed around.

**Comment by cyclowns** [comment]
@ExcessiveUseOfCobblestone 
you seem to be under the impression that making anything higher than a maxcap is black magic that you can only figure out how to do after ingesting every single line of code in the TG codebase anally.

in reality, people that have actually codedived to a significant extent and understand bombs very well (me or @Whoneedspacee for example) pretty much just did it because
1. we like math/thermodynamics
2. we like minmaxing
3. and we like atmos

4500 LR (the max for my formula) bombs really aren't that hard to understand. where it gets hard is you when you start really only caring about theoreticals and get to things like 308000 LR.

**Comment by cyclowns** [comment]
another note
if you spent more than 30 seconds looking at it, you might notice that the formula i presented ALSO doesn't cater to people who only care about theoreticals (i.e bombs requiring fusion, hyper-noblium, stimulum, insane amounts of trit). its just that, you know, mine is based on how the progression of bombs actually works.

**Comment by cyclowns** [comment]
ANOTHER note
balancing max points around literally bottom of the barrel players and bombs is the exact opposite way to get people to explore the mechanic. because with your formula its INSANELY easy to get 50k points, what incentive is there to even attempt to experiment beyond that besides theoreticals? having uncapped points (or at least a higher cap with respect to LR, like what my formula suggests) helped get a lot of people into toxins because they would see people get insanely high point bombs. my formula still simulates that to a certain extent, but without completely ruining techweb balance by giving too many points.

**Comment by cyclowns** [comment]
ANOTHER NOTE COMBO x 3

people resorting to codediving or youtube videos to figure out larger bombs isn't toxins fault, it's the wikis fault for not giving people accurate or enough information so that they can find out The Joy of Toxins (hosted by Bomb Ross) by themselves

which, coincidentally, is something that me, @Whoneedspacee, and @granpawalton are thinking about trying to fix

**Comment by ExcessiveUseOfCobblestone** [comment]
Alright, if you can turn your formula into a byond approved formula I will use that instead and give you credit.

**Comment by Grazyn** [comment]
All this math talk still isn't addressing the issue of science starting with tens of thousands more tech points than they're supposed to.

Because experimenting with toxins is all fun and everything, but let's be honest here, scientists will still rush to make the quickest meta bomb they can make at roundstart to get that 25k/50k/75k/100k burst of points the new formula will allow them to. This still breaks the balance of techwebs as originally designed.

**Comment by ExcessiveUseOfCobblestone** [comment]
Techwebs passive gen is balanced around being nearly nil input from anything (hence why it's lmao'd right now so easily). As more active ways are introduced, passive gen is to go down to compensate so really you're forced to "rush" (or, more appropriately, do your job).

The cap is still 50K, it's just that with the new formula it takes a considerable amount of work ("rushing" would be doing it the entire shift from the feedback) to get to that point. As with the old formula, bombs that surpass the cap after it's been reached will be given 1K.

The only time it's throwing everything out of whack would be while passive gen remains the same in light of various active methods of pointgain, which has yet to be achieved.

**Comment by Whoneedspacee** [comment]
The max on this formula is about 200k before the update so that's a good 15-20 minutes, much more than what the meta was previously and giving out much less in return.

**Comment by Dorsisdwarf** [comment]
!()[http://puu.sh/BuLyU/80bda731e7.png]

why take this away from us

**Comment by Barhandar** [comment]
Mining doesn't need any of your bombs, they have their own bombs _and_ research gear is far better anyway. Instead of this hardcap I suggest specific softcap, curved scaling after that (so regular bombs would do useful gains, and current tritium bombs that make 200k points would only do 40k, for example) and _no diminishing returns on followup explosions_. So it becomes a choice between exploding all TTVs for research or going Cuban Pete with them instead.

Hardcap + diminishing returns just means toxins _still_ explodes just one bomb and then fucks off.

**Comment by Whoneedspacee** [comment]
> Mining doesn't need any of your bombs, they have their own bombs _and_ research gear is far better anyway. Instead of this hardcap I suggest specific softcap, curved scaling after that (so regular bombs would do useful gains, and current tritium bombs that make 200k points would only do 40k, for example) and _no diminishing returns on followup explosions_. So it becomes a choice between exploding all TTVs for research or going Cuban Pete with them instead.
> 
> Hardcap + diminishing returns just means toxins _still_ explodes just one bomb and then fucks off.

You're a bit late to the discussion, it's pretty set in stone, you need to get like a 300k ish bomb to reach this current cap anyways.

**Comment by deathride58** [comment]
You can still 100% RnD with toxins with this PR if cargo can scrounge up the necessary 438,000 cargo points to order a grand total of 145 extra TTVs. There's 55k potential research points max from the roundstart TTVs, plus 1k points per TTVs. I'm assuming an rnd completion point total of 200k for the extra TTVs for completion calculation

It might be worth considering upping the maximum point generation post-softcap to 2.5k points from the original 1k points, and increasing the cost of TTV crates from 6k cargo points to 15k cargo points. This would increase the potential research point max of roundstart TTVs to 62.5k, and decrease the extra TTVs required for RnD completion to 55 extra TTVs (420,000 cargo points). This would give the TTV crate an actual use for once outside of antag cargo tech gameplay, as scientists would actually have a legitimate reason to request extra TTV crates from cargo. Making all of those TTVs would still be very tedious, yes, but it'll provide more than enough incentive to promote cooperation between cargo and science.

**Comment by adamkad1** [comment]
worst pr ever

**Comment by RogueSteampunker** [comment]
So basically, we're back where we were pre-techwebs, where any use of toxins outside of blowing the station to high hell is a niche use that is basically useless in the long and short of it. Lovely.

**Comment by optimumtact** [comment]
Yes we made toxins take more than 15 minutes at roundstart to get enough points to complete the entire tech tree and it is now useless obviously, whatever will you do now that the first 15 minutes of your round can't complete techwebs and you'll have to make a lot more bombs instead?

**Comment by blobbernaut** [comment]
If this wasn't a blatant grudge PR because you hate somebody, you'd have just adjusted the scaling formula instead of capping it like this. All THIS does is remove ANY incentive to make a giant bomb. Now go back, and just change it from, like, size^0.5 to like, size^0.4 or something. 

Or do it like tax brackets and decrease the scaling factor after it hits 50,000. You know, reward people for knowing what they're doing. LIKE EVERYTHING ELSE IN THE GAME DOES.

**Comment by octareenroon91** [comment]
>Or do it like tax brackets and decrease the scaling factor after it hits 50,000. You know, reward people for knowing what they're doing. LIKE EVERYTHING ELSE IN THE GAME DOES.

Seems I didn't say "use a piecewise function instead" before now. I had meant to.

**Comment by ExcessiveUseOfCobblestone** [comment]
> grudge PR 

Sure? I have a "grudge" against the ability to nullify the progression system. The old formula was made when just plas/oxy bombs existed so really this was just an oversight fix.

The formula was changed by cyclown and supported by the toxins brigatde to encourage giant bomb. OP image is old and doesn't reflect the formuler' in code.

After 50K points you get a flat 1K point increase. IIRC getting to 50K is like 4500 Light Radius so good luck :)

**Pull request merged**